src/test/java/io/vertx/test/core/HttpTest.java

```diff
@@ -129,6 +131,40 @@ public void testClientChaining() {
     await();
   }
 
+  @Test
+  public void testListenSocketAddress() {
+    NetClient netClient = vertx.createNetClient();
+    server = vertx.createHttpServer().requestHandler(req -> req.response().end());
+    SocketAddress sockAddress = SocketAddress.inetSocketAddress(DEFAULT_HTTP_PORT, DEFAULT_HTTP_HOST);
+    server.listen(sockAddress, onSuccess(server -> {
+      netClient.connect(sockAddress, onSuccess(sock -> {
+        sock.handler(buf -> testComplete());
+        sock.write("GET / HTTP/1.1\r\n");
+      }));
+    }));
+
+    await();
+  }
+
+  @Test
+  public void testListenDomainSocketAddress() throws Exception {
+    Vertx vx = Vertx.vertx(new VertxOptions().setPreferNativeTransport(true));
```

Comment:
you need to close the created vertx in a finally block otherwise we leak resources in tests